app/services/fixed_charges/update_service.rb

```diff
@@ -19,7 +19,7 @@ def call
         # Note: when updating a fixed_charge, we can't update pay_in_advance and prorated,
         fixed_charge.charge_model = params[:charge_model] unless plan.attached_to_subscriptions?
         fixed_charge.invoice_display_name = params[:invoice_display_name] unless cascade
-        fixed_charge.units = params[:units]
+        fixed_charge.units = params[:units] if params.key?(:units)
```

Comment:
Ah, I had only one comment here:
right now for charges we do not have this check is params contain this pair, so we always assume that if fixed_charge is updated, the whole payload will be sent. 
If we're adding a possibility to send not the whole payload, it might make sense to do the same check for `charge_model`, `invoice_display_name`...